src/unix/apple/macos/ffi.rs

```diff
@@ -322,82 +328,62 @@ mod io_service {
 
     #[cfg(not(feature = "apple-sandbox"))]
     extern "C" {
-        pub fn IOHIDEventSystemClientCreate(allocator: CFAllocatorRef)
-            -> IOHIDEventSystemClientRef;
+        pub fn IOHIDEventSystemClientCreate(
+            allocator: Option<&CFAllocator>,
+        ) -> Option<NonNull<IOHIDEventSystemClient>>;
 
         pub fn IOHIDEventSystemClientSetMatching(
-            client: IOHIDEventSystemClientRef,
-            matches: CFDictionaryRef,
+            client: &IOHIDEventSystemClient,
```

Comment:
I'm surprised that it works. Are references in FFI functions converted to pointers?

Reply:
Yes, see [the docs on ABI compatibility](https://doc.rust-lang.org/1.84.0/std/primitive.fn.html#abi-compatibility) (which is a stronger guarantee than just having the same size and alignment).

Reply:
TIL, thanks a lot for the link!